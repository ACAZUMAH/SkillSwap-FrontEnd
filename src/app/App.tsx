import { ApolloProvider } from "@apollo/client";
import type React from "react";
import { Provider } from "react-redux";
import { PersistGate } from "redux-persist/integration/react";
import { MantineThemeProvider } from "src/providers";
import { persistor, store } from "src/redux/store";
import { AppRouter } from "src/router/router";
import { client } from "src/services/grapql-client";

const App: React.FC = () => {
  return (
    <>
      <Provider store={store}>
        <PersistGate loading={null} persistor={persistor}>
          <ApolloProvider client={client}>
            <MantineThemeProvider>
              <AppRouter />
            </MantineThemeProvider>
          </ApolloProvider>
        </PersistGate>
      </Provider>
    </>
  );
};

export default App;
