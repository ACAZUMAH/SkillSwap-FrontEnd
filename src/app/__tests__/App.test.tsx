import { render } from "@testing-library/react";
import App from "../App";
import { BrowserRouter } from "react-router-dom";

test("renders the app landing page", () => {
  window.history.pushState({}, "Test Page", "/");
  const { asFragment } = render(
    <BrowserRouter>
      <App />
    </BrowserRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
