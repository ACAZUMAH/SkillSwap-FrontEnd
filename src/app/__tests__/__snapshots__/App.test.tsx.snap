// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders the app landing page 1`] = `
<DocumentFragment>
  <style
    data-mantine-styles="true"
  >
    :root{--mantine-primary-color-filled: var(--mantine-color-brand-filled);--mantine-primary-color-filled-hover: var(--mantine-color-brand-filled-hover);--mantine-primary-color-light: var(--mantine-color-brand-light);--mantine-primary-color-light-hover: var(--mantine-color-brand-light-hover);--mantine-primary-color-light-color: var(--mantine-color-brand-light-color);--mantine-primary-color-0: var(--mantine-color-brand-0);--mantine-primary-color-1: var(--mantine-color-brand-1);--mantine-primary-color-2: var(--mantine-color-brand-2);--mantine-primary-color-3: var(--mantine-color-brand-3);--mantine-primary-color-4: var(--mantine-color-brand-4);--mantine-primary-color-5: var(--mantine-color-brand-5);--mantine-primary-color-6: var(--mantine-color-brand-6);--mantine-primary-color-7: var(--mantine-color-brand-7);--mantine-primary-color-8: var(--mantine-color-brand-8);--mantine-primary-color-9: var(--mantine-color-brand-9);--mantine-color-brand-0: #e5edff;--mantine-color-brand-1: #c1d3ff;--mantine-color-brand-2: #9ab8ff;--mantine-color-brand-3: #709eff;--mantine-color-brand-4: #4a85ff;--mantine-color-brand-5: #2b6ef7;--mantine-color-brand-6: #1f5de5;--mantine-color-brand-7: #1f54f2;--mantine-color-brand-8: #1c47c2;--mantine-color-brand-9: #193ca1;}:root[data-mantine-color-scheme="dark"]{--mantine-color-anchor: var(--mantine-color-brand-4);--mantine-color-dark-filled: var(--mantine-color-dark-6);--mantine-color-dark-filled-hover: var(--mantine-color-dark-7);--mantine-color-dark-light: rgba(66, 66, 66, 0.15);--mantine-color-dark-light-hover: rgba(66, 66, 66, 0.2);--mantine-color-dark-light-color: var(--mantine-color-dark-1);--mantine-color-dark-outline: var(--mantine-color-dark-2);--mantine-color-dark-outline-hover: rgba(130, 130, 130, 0.05);--mantine-color-gray-filled: var(--mantine-color-gray-6);--mantine-color-gray-filled-hover: var(--mantine-color-gray-7);--mantine-color-gray-light: rgba(206, 212, 218, 0.15);--mantine-color-gray-light-hover: rgba(206, 212, 218, 0.2);--mantine-color-gray-light-color: var(--mantine-color-gray-1);--mantine-color-gray-outline: var(--mantine-color-gray-2);--mantine-color-gray-outline-hover: rgba(233, 236, 239, 0.05);--mantine-color-red-filled: var(--mantine-color-red-6);--mantine-color-red-filled-hover: var(--mantine-color-red-7);--mantine-color-red-light: rgba(255, 135, 135, 0.15);--mantine-color-red-light-hover: rgba(255, 135, 135, 0.2);--mantine-color-red-light-color: var(--mantine-color-red-1);--mantine-color-red-outline: var(--mantine-color-red-2);--mantine-color-red-outline-hover: rgba(255, 201, 201, 0.05);--mantine-color-pink-filled: var(--mantine-color-pink-6);--mantine-color-pink-filled-hover: var(--mantine-color-pink-7);--mantine-color-pink-light: rgba(247, 131, 172, 0.15);--mantine-color-pink-light-hover: rgba(247, 131, 172, 0.2);--mantine-color-pink-light-color: var(--mantine-color-pink-1);--mantine-color-pink-outline: var(--mantine-color-pink-2);--mantine-color-pink-outline-hover: rgba(252, 194, 215, 0.05);--mantine-color-grape-filled: var(--mantine-color-grape-6);--mantine-color-grape-filled-hover: var(--mantine-color-grape-7);--mantine-color-grape-light: rgba(218, 119, 242, 0.15);--mantine-color-grape-light-hover: rgba(218, 119, 242, 0.2);--mantine-color-grape-light-color: var(--mantine-color-grape-1);--mantine-color-grape-outline: var(--mantine-color-grape-2);--mantine-color-grape-outline-hover: rgba(238, 190, 250, 0.05);--mantine-color-violet-filled: var(--mantine-color-violet-6);--mantine-color-violet-filled-hover: var(--mantine-color-violet-7);--mantine-color-violet-light: rgba(151, 117, 250, 0.15);--mantine-color-violet-light-hover: rgba(151, 117, 250, 0.2);--mantine-color-violet-light-color: var(--mantine-color-violet-1);--mantine-color-violet-outline: var(--mantine-color-violet-2);--mantine-color-violet-outline-hover: rgba(208, 191, 255, 0.05);--mantine-color-indigo-filled: var(--mantine-color-indigo-6);--mantine-color-indigo-filled-hover: var(--mantine-color-indigo-7);--mantine-color-indigo-light: rgba(116, 143, 252, 0.15);--mantine-color-indigo-light-hover: rgba(116, 143, 252, 0.2);--mantine-color-indigo-light-color: var(--mantine-color-indigo-1);--mantine-color-indigo-outline: var(--mantine-color-indigo-2);--mantine-color-indigo-outline-hover: rgba(186, 200, 255, 0.05);--mantine-color-blue-filled: var(--mantine-color-blue-6);--mantine-color-blue-filled-hover: var(--mantine-color-blue-7);--mantine-color-blue-light: rgba(77, 171, 247, 0.15);--mantine-color-blue-light-hover: rgba(77, 171, 247, 0.2);--mantine-color-blue-light-color: var(--mantine-color-blue-1);--mantine-color-blue-outline: var(--mantine-color-blue-2);--mantine-color-blue-outline-hover: rgba(165, 216, 255, 0.05);--mantine-color-cyan-filled: var(--mantine-color-cyan-6);--mantine-color-cyan-filled-hover: var(--mantine-color-cyan-7);--mantine-color-cyan-light: rgba(59, 201, 219, 0.15);--mantine-color-cyan-light-hover: rgba(59, 201, 219, 0.2);--mantine-color-cyan-light-color: var(--mantine-color-cyan-1);--mantine-color-cyan-outline: var(--mantine-color-cyan-2);--mantine-color-cyan-outline-hover: rgba(153, 233, 242, 0.05);--mantine-color-teal-filled: var(--mantine-color-teal-6);--mantine-color-teal-filled-hover: var(--mantine-color-teal-7);--mantine-color-teal-light: rgba(56, 217, 169, 0.15);--mantine-color-teal-light-hover: rgba(56, 217, 169, 0.2);--mantine-color-teal-light-color: var(--mantine-color-teal-1);--mantine-color-teal-outline: var(--mantine-color-teal-2);--mantine-color-teal-outline-hover: rgba(150, 242, 215, 0.05);--mantine-color-green-filled: var(--mantine-color-green-6);--mantine-color-green-filled-hover: var(--mantine-color-green-7);--mantine-color-green-light: rgba(105, 219, 124, 0.15);--mantine-color-green-light-hover: rgba(105, 219, 124, 0.2);--mantine-color-green-light-color: var(--mantine-color-green-1);--mantine-color-green-outline: var(--mantine-color-green-2);--mantine-color-green-outline-hover: rgba(178, 242, 187, 0.05);--mantine-color-lime-filled: var(--mantine-color-lime-6);--mantine-color-lime-filled-hover: var(--mantine-color-lime-7);--mantine-color-lime-light: rgba(169, 227, 75, 0.15);--mantine-color-lime-light-hover: rgba(169, 227, 75, 0.2);--mantine-color-lime-light-color: var(--mantine-color-lime-1);--mantine-color-lime-outline: var(--mantine-color-lime-2);--mantine-color-lime-outline-hover: rgba(216, 245, 162, 0.05);--mantine-color-yellow-filled: var(--mantine-color-yellow-6);--mantine-color-yellow-filled-hover: var(--mantine-color-yellow-7);--mantine-color-yellow-light: rgba(255, 212, 59, 0.15);--mantine-color-yellow-light-hover: rgba(255, 212, 59, 0.2);--mantine-color-yellow-light-color: var(--mantine-color-yellow-1);--mantine-color-yellow-outline: var(--mantine-color-yellow-2);--mantine-color-yellow-outline-hover: rgba(255, 236, 153, 0.05);--mantine-color-orange-filled: var(--mantine-color-orange-6);--mantine-color-orange-filled-hover: var(--mantine-color-orange-7);--mantine-color-orange-light: rgba(255, 169, 77, 0.15);--mantine-color-orange-light-hover: rgba(255, 169, 77, 0.2);--mantine-color-orange-light-color: var(--mantine-color-orange-1);--mantine-color-orange-outline: var(--mantine-color-orange-2);--mantine-color-orange-outline-hover: rgba(255, 216, 168, 0.05);--mantine-color-brand-text: var(--mantine-color-brand-4);--mantine-color-brand-filled: var(--mantine-color-brand-6);--mantine-color-brand-filled-hover: var(--mantine-color-brand-7);--mantine-color-brand-light: rgba(74, 133, 255, 0.15);--mantine-color-brand-light-hover: rgba(74, 133, 255, 0.2);--mantine-color-brand-light-color: var(--mantine-color-brand-1);--mantine-color-brand-outline: var(--mantine-color-brand-2);--mantine-color-brand-outline-hover: rgba(154, 184, 255, 0.05);}:root[data-mantine-color-scheme="light"]{--mantine-color-anchor: var(--mantine-color-brand-6);--mantine-color-brand-text: var(--mantine-color-brand-filled);--mantine-color-brand-filled: var(--mantine-color-brand-6);--mantine-color-brand-filled-hover: var(--mantine-color-brand-7);--mantine-color-brand-light: rgba(31, 93, 229, 0.1);--mantine-color-brand-light-hover: rgba(31, 93, 229, 0.12);--mantine-color-brand-light-color: var(--mantine-color-brand-6);--mantine-color-brand-outline: var(--mantine-color-brand-6);--mantine-color-brand-outline-hover: rgba(31, 93, 229, 0.05);}
  </style>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
</DocumentFragment>
`;
